RSA.xs

```diff
@@ -55,16 +86,22 @@ void croakSsl(char* p_file, int p_line)
 
 char _is_private(rsaData* p_rsa)
 {
-    const BIGNUM *d;
 #if OLD_CRUFTY_SSL_VERSION
+    const BIGNUM* d;
     d = p_rsa->rsa->d;
 #else
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+    BIGNUM* d = NULL;
+    EVP_PKEY_get_bn_param(p_rsa->rsa, OSSL_PKEY_PARAM_RSA_D, &d);
```

Comment:
`EVP_get_bn_param()` allocates `d`, so this will need to be freed before you return (since this is a copy of the private key you should use `BN_clear_free(d)`).

```
     EVP_PKEY_get_bn_param() retrieves a key pkey BIGNUM value **bn associated
     with a name of key_name. If *bn is NULL then the BIGNUM is allocated by
     the method.
```